src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -362,6 +362,83 @@ seastar::future<std::tuple<Status, CollectionCreateResponse>> HTTPProxy::_handle
     });
 }
 
+seastar::future<std::tuple<k2::Status, CreateQueryResponse>> HTTPProxy::_handleCreateQuery(CreateQueryRequest&& request) {
+    return _client.createQuery(request.collectionName, request.schemaName)
+    .then([this, req=std::move(request)] (auto&& result) mutable {
+        if(!result.status.is2xxOK()) {
+            return RPCResponse(std::move(result.status), CreateQueryResponse{0});
+        }
+        K2LOG_D(log::httpproxy, "begin query {}", result);
+        if (!req.startScanRecord.is_null()) {
+            serializeRecordFromJSON(result.query.startScanRecord, std::move(req.startScanRecord));
+        }
+        if (!req.endScanRecord.is_null()) {
+            serializeRecordFromJSON(result.query.endScanRecord, std::move(req.endScanRecord));
+        }
+        if (req.limit != 0) {
+            result.query.setLimit(req.limit);
+        }
+        if (req.reverse) {
+            result.query.setReverseDirection(req.reverse);
+        }
+        _queries[_queryID++] = std::move(result.query);
```

Comment:
We should have timeouts associated with these entries and  we should stop tracking them if timeout is reached. Same for transactions i think.

Reply:
Created issue https://github.com/futurewei-cloud/chogori-platform/issues/286 to handle it